tests/kernel/device/src/mmio_multireg.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ * Copyright (c) 2022 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr/device.h>
+
+#define DT_DRV_COMPAT	fakedriver_multireg
+
+/*
+ * Driver with multiple MMIO regions to manage defined into DT
+ */
+
+struct foo_multireg_dev_data {
+	int baz;
+
+	DEVICE_MMIO_NAMED_RAM(chip);
+	DEVICE_MMIO_NAMED_RAM(dale);
+};
+
+struct foo_multireg_dev_data foo_multireg_data;
+
+struct foo_multireg_config_info {
+	DEVICE_MMIO_NAMED_ROM(chip);
+	DEVICE_MMIO_NAMED_ROM(dale);
+};
+
+const struct foo_multireg_config_info foo_multireg_config = {
+	DEVICE_MMIO_NAMED_ROM_INIT_BY_NAME(chip, DT_DRV_INST(0)),
+	DEVICE_MMIO_NAMED_ROM_INIT_BY_NAME(dale, DT_DRV_INST(0))
+};
+
+#define DEV_DATA(dev)	((struct foo_multireg_dev_data *)((dev)->data))
+#define DEV_CFG(dev)	((struct foo_multireg_config_info *)((dev)->config))
```

Comment:
IMO, these implicit macro requirements are a little bit hard to use and easy to forget. Fortunately, the compiler will alert:)

Reply:
Agree, but at least it is clearly documented (somewhere).